ut/performance_tests.cpp

```diff
@@ -0,0 +1,233 @@
+    // comparing to 8M bytes of serialized data for ColumnFixedString and ColumUInt64.
+    // So in order to make comparison mode fair, reducing size of data item.
+    return generate_string_view<7>(index);
+}
+
+std::string_view generate(const ColumnFixedString&, size_t index)
+{
+    return generate_string_view<8>(index);
+}
+
+std::string_view generate(const ColumnLowCardinalityT<ColumnString>&, size_t index)
+{
+    return generate_string_view<7>(index);
+}
+
+std::string_view generate(const ColumnLowCardinalityT<ColumnFixedString>&, size_t index)
+{
+    return generate_string_view<8>(index);
+}
+
+template <typename ColumnType>
+auto ValidateColumnItems(const ColumnType & col, size_t expected_items)
+{
+    ASSERT_EQ(expected_items, col.Size());
+    // validate that appended items match expected
+    for (size_t i = 0; i < expected_items; ++i)
+    {
+        SCOPED_TRACE(i);
+
+        ASSERT_EQ(col.At(i), generate(col, i));
+        ASSERT_EQ(col[i], generate(col, i));
+    }
+};
+
+template <typename ColumnType>
+ColumnType InstantiateColumn() {
+    if constexpr (std::is_same_v<ColumnType, ColumnFixedString>) {
+        return ColumnType(8);
+    } else if constexpr (std::is_same_v<ColumnType, ColumnLowCardinalityT<ColumnFixedString>>) {
+        return ColumnType(8);
+    } else {
+        return ColumnType();
+    }
+}
+
+template <typename ColumnType>
+class ColumnPerformanceTest : public ::testing::Test {
+};
+
+TYPED_TEST_CASE_P(ColumnPerformanceTest);
+
+TYPED_TEST_P(ColumnPerformanceTest, SaveAndLoad) {
+    auto column = InstantiateColumn<TypeParam>();
+
+    const size_t ITEMS_COUNT = 1'000'000;
+    const int LOAD_AND_SAVE_REPEAT_TIMES = 10; // run Load() and Save() multiple times to cancel out measurement errors.
+
+    std::cerr << "\n===========================================================" << std::endl;
+    std::cerr << "\t" << ITEMS_COUNT << " items of " << column.Type()->GetName()  << std::endl;
+
+    PausableTimer<std::chrono::microseconds> timer;
+
+    timer.Start();
+    for (size_t i = 0; i < ITEMS_COUNT; ++i)
+    {
+        const auto value = generate(column, i);
+        column.Append(value);
+    }
+
+    EXPECT_EQ(ITEMS_COUNT, column.Size());
+    std::cerr << "Appending:\t" << timer.GetTotalElapsed().count() << " us"
+              << std::endl;
+
+    timer.Reset();
+    timer.Start();
+    EXPECT_NO_FATAL_FAILURE(ValidateColumnItems(column, ITEMS_COUNT));
+    std::cerr << "Accessing (twice):\t" << timer.GetTotalElapsed().count() << " us"
+              << std::endl;
+
+    Buffer buffer;
+
+    // Save
+    {
+        timer.Reset();
+
+        for (int i = 0; i < LOAD_AND_SAVE_REPEAT_TIMES; ++i)
+        {
+            buffer.clear();
+            BufferOutput bufferOutput(&buffer);
+            CodedOutputStream ostr(&bufferOutput);
+
+            timer.Start();
+            column.Save(&ostr);
+            ostr.Flush();
```

Comment:
Do you really want to measure flush too?

Reply:
Yep, since lots of interesting things can happen there, also I am measuring the whole process, the way it is in real cases.

Reply:
OK, however potentially LowCardinality + flush, and regular type + flush numbers may differ insignificantly, if flush is heavy. While isolated LowCardinality may be like several times slower than a regular type, and that difference just dwarfed by flush overhead.

Reply:
It is quite the opposite: serialization/deserialization of `LowCardinality` columns is **MUCH** faster (up to 100 times faster than bare `String` and up to 3 times faster than bare `FixedSring` columns)

Reply:
Yes, sorry, my bad. But the idea remains the same, observer wont be able so properly see that degree of difference.